src/engine/movementCalculator.ts

```diff
@@ -0,0 +1,32 @@
+import Square from "./square";
+import Board from "./board";
+import Piece from "./pieces/piece";
+import King from "./pieces/king";
+
+export default class MovementCalculator {
+
+    public static checkInDirection(piece: Piece, moves: Square[], board: Board, xDir: number, yDir: number): void {
+        let currentSquare: Square = board.findPiece(piece);
+
+        let x: number = currentSquare.col + xDir;
+        let y: number = currentSquare.row + yDir;
+        while (Board.positionsExists(y, x)) {
+            let hitPiece: Piece | undefined;
+            hitPiece = board.getPiece(Square.at(y, x));

```

Comment:
Poate fi initializat din in momentul declararii. Avand in vedere ca este redeclarata la fiecare pas din while, poate fi const pentru ca nu se modifica pe parcursul pasului curent.